Docs: Update to reference-style markdown links

Crosse opened the pull request:
As per https://github.com/coreos/docs/blob/master/STYLE.md,
reference-style links should be used in documentation. This converts all
documentation to use this format instead of inline links.

Inline links that refer to anchors in the same file were left unchanged.

This is part one of a larger task where I'd like to work on updating the documentation to abide by the CoreOS style guide, as well as generally clean up the docs grammar- and content-wise, etc. If you're not interested in this type of bulk documentation fix, let me know! I know some projects would rather not update docs unless the content itself needs to change.

lucab commented:
Paging @joshix et al. for a review here.

> If you're not interested in this type of bulk documentation fix, let me know!

We typically welcome any kind of fixes, including bulk documentation changes if they make sense and improve uniformity! For this specific PR, I'm deferring to our docops friends for the final word.

> I know some projects would rather not update docs unless the content itself needs to change.

We don't have such a policy. On the opposite side, some of our docs may be in need of a bit of love, so contributions are much appreciated.

Crosse commented:
Sounds good! I forgot to mention that this patch fixes some links that used the full github URL to the markdown doc, which would "break" the coreos.com documentation library links (i.e., instead of linking to the generated HTML file, it would link back to the github doc).

I'll wait for @joshix to weigh in on this and go from there.

joshix commented:
@Crosse Thanks for this! We'll get someone on reviewing it.

Could you rebase this to resolve the conflict in Documentation/hacking?

Documentation fixes are welcome and a good way to get on the ground running as a contributor, since you can learn a bit about the software even while applying style fixes and the like.

Crosse commented:
Okay, I've rebased my branch and fixed an inline-style link that was introduced to hacking.md in that last commit.

(As an aside: after I rebase and fix stuff, I'm doing a "git push -f" to update my remote branch. Is that what I should be doing? I'm a little shaky how to work with PRs.)

joshix commented:
@Crosse Thank you for rebasing. Yes, you appear to be doing it correctly. We like commits squashed into reasonable units, with a rebase push to your fork and branch, as you've done here.

ElijahCaine commented on Documentation/networking/overview.md:
```diff
@@ -341,13 +341,27 @@ As a reminder, the sort order of the loaded networks is detailed in the chapter
+[dns]: dns.md
+[examples-bridge]: examples-bridge.md
+[ipvlan]: https://www.kernel.org/doc/Documentation/networking/ipvlan.txt
+[ipvlan-dupv6]: http://thread.gmane.org/gmane.linux.network/363346/focus=363345
```
This link is also broken. Might want to get rid of it.

Crosse replied:
I wasn't sure what to do with this. Gmane says they're still bringing all their old content online, but gmane.linux.network isn't there yet and I wasn't able to find another source for this message. I'll just remove the link entirely, I think.

Crosse commented on Documentation/app-container.md:
```diff
@@ -37,10 +37,16 @@ To validate that `rkt` successfully implements the ACE part of the spec, use the
+[docker2aci]: https://github.com/appc/docker2aci
+[goaci]: https://github.com/appc/goaci
+[k8s-pods]: http://kubernetes.io/docs/user-guide/pods/
+[running-docker-images]: running-docker-images.md
```
I tested these links again and they all worked for me... is it the running-docker-images link that's broken?

Crosse commented on Documentation/app-environment.md:
```diff
@@ -22,3 +22,5 @@ Since rkt v1.2.0, rkt gives access to systemd-journald's sockets in the /run/sys
 #### /dev/log
 
 Since rkt v1.2.0, if /dev/log does not exist in the image, it will be created as a symlink to /run/systemd/journal/dev-log.
+[networking-dns]: networking/dns.md
```
I tested this link again and it worked for me. I'm not sure what is going on here.

Crosse commented:
@ElijahCaine I checked the links you said were broken, but they all worked for me, so I'm not sure what's going on there. I rebased and removed the dead Gmane link in networking/overview.md.

Could you verify whether those other links are still broken for you?

ElijahCaine commented:
@Crosse This looks good to me.  I'm not sure why that link was broken before, but it was probably just something on GitHub's end. 👍 from me. Last thoughts @joshix ?

joshix commented:
Thanks, folks!
